rotkehlchen/chain/ethereum/modules/eth2/structures.py

```diff
@@ -63,132 +64,56 @@ def serialize(self) -> dict[str, Any]:
 ValidatorDailyStatsDBTuple = tuple[
     int,  # validator index
     int,  # timestamp
-    str,  # usd_price_start
-    str,  # usd_price_end
     str,  # pnl_amount
-    str,  # start_balance
-    str,  # end_balance
-    int,  # missed_attestations
-    int,  # orphaned_attestations
-    int,  # proposed_blocks
-    int,  # missed_blocks
-    int,  # orphaned_blocks
-    int,  # included_attester_slashings
-    int,  # proposer_attester_slashings
-    int,  # deposits_number
-    str,  # amount_deposited
 ]
 
 
 @dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
 class ValidatorDailyStats(AccountingEventMixin):
     validator_index: int  # keeping the index here so it can be shown in accounting
     timestamp: Timestamp
-    start_usd_price: FVal = ZERO
-    end_usd_price: FVal = ZERO
     pnl: FVal = ZERO
-    start_amount: FVal = ZERO
-    end_amount: FVal = ZERO
-    missed_attestations: int = 0
-    orphaned_attestations: int = 0
-    proposed_blocks: int = 0
-    missed_blocks: int = 0
-    orphaned_blocks: int = 0
-    included_attester_slashings: int = 0
-    proposer_attester_slashings: int = 0
-    deposits_number: int = 0
-    amount_deposited: FVal = ZERO
     ownership_percentage: FVal = ONE  # customized by get_eth2_history_events
 
     def __str__(self) -> str:
         return f'ETH2 validator {self.validator_index} daily stats'
 
     @property
     def pnl_balance(self) -> Balance:
-        usd_price = (self.start_usd_price + self.end_usd_price) / 2
+        try:
+            usd_price = PriceHistorian().query_historical_price(
+                from_asset=A_ETH,
+                to_asset=A_USD,
+                timestamp=self.timestamp,
+            )
+        except NoPriceForGivenTimestamp:
+            usd_price = Price(ZERO)
+
         return Balance(
             amount=self.pnl,
             usd_value=self.pnl * usd_price,
         )
 
-    @property
-    def start_balance(self) -> Balance:
```

Comment:
wonder why we were doing this before. Only thing I can think of is to completely avoid any oracle queries and be able to immediately return amount + usd value? But still if the oracle query is cached in the DB once it's then saved and we got it forever, right?

Reply:
> Only thing I can think of is to completely avoid any oracle queries and be able to immediately return amount + usd value?

Yeah, could be.

> But still if the oracle query is cached in the DB once it's then saved and we got it forever, right?

Yes.